econml/automated_ml.py

```diff
@@ -0,0 +1,360 @@
+    def __init__(self, automl_config, workspace,
+                 experiment_name_prefix="aml_experiment"):
+        self._show_output = automl_config._show_output
+        self._workspace = workspace
+        self._automl_config = automl_config
+        self._experiment_name_prefix = experiment_name_prefix
+
+    def get_params(self, deep=True):
+        # Must be implemented for MultiOutputRegressor to view _InnerAutomatedMLModel
+        # as an sklearn estimator
+        return {
+            'workspace': self._workspace,
+            'automl_config': self._automl_config,
+            'experiment_name_prefix': self._experiment_name_prefix
+        }
+
+    def fit(self, X, y, sample_weight=None):
+        # fit implementation for a single output model.
+        # Create experiment for specified workspace
+        automl_config = copy.deepcopy(self._automl_config)
+        current_time = time.localtime()
+        current_time_string = time.strftime('%y_%m_%d-%H_%M_%S', current_time)
+        experiment_name = self._experiment_name_prefix + "_" + current_time_string
+        self._experiment = Experiment(self._workspace, experiment_name)
+        # Configure automl_config with training set information.
+        automl_config.user_settings['X'] = X
+        automl_config.user_settings['y'] = y
+        automl_config.user_settings['sample_weight'] = sample_weight
+        # Wait for remote run to complete, the set the model
+        print("Experiment " + experiment_name + " has started.")
+        local_run = self._experiment.submit(automl_config, show_output=self._show_output)
+        print("Experiment " + experiment_name + " completed.")
+        _, self._model = local_run.get_output()
+
+    def predict(self, X):
+        return self._model.predict(X)
+
+    def predict_proba(self, X):
+        return self._model.predict_proba(X)
+
+
+class AutomatedMLMixin():
+    def __init__(self, *args, **kwargs):
+        """
+        Mixin enabling users to leverage automatedML as their model of choice in
+        Double Machine Learners and Doubly Robust Learners. It instantiates
+        AutomatedMLModels for each automl_config provided and pass them as
+        parameters into its parent class.
+
+        Parameters
+        ----------
+        args: List, optional
+           args that are passed in order to initiate the final automatedML run.
+           Any arg, that is an AutoMLConfig, will be converted into as
+           AutomatedMLModel.
+
+        kwargs: Dict, optional
+           kwargs that are passed in order to initiate the final automatedML run.
+           Any kwarg, that is an AutoMLConfig, will be converted into as
+           AutomatedMLModel.
+       """
+        # Loop through the kwargs and args if any of them is an AutoMLConfig file, pass them
+        # create model and pass model into final.
+        new_args = ()
+        for var in args:
+            new_args += (var,) if var is not None else (None,)
```

Comment:
1. It doesn't look like the check for `None` accomplishes anything.
2. Shouldn't positional arguments also be wrapped via `_get_automated_ml_model`?